api/v1alpha1/proxyplugin_types.go

```diff
@@ -0,0 +1,73 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// OpenShiftRouteTarget captures the look up information for retrieving an OpenShift Route object in the member cluster.
+type OpenShiftRouteTarget struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+// ProxyPluginSpec defines the desired state of ProxyPlugin
+// +k8s:openapi-gen=true
+type ProxyPluginSpec struct {
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// OpenShiftRouteTargetEndpoint is an optional field that represents the look up information for an OpenShift Route
+	// as the endpoint for the registration service to proxy requests to that have the https://<proxy-host>/plugins/<ProxyPlugin.ObjectMeta.Name>
+	// in its incoming URL.  As we add more types besides OpenShift Routes, we will add more optional fields to this spec
+	// object
```

Comment:
To be honest it's still a bit unclear to me what is the proper way of defining optional fields. But I personally always add `// +optional` to optional field. Not sure if it's really required if it's a pointer... ;) According to https://github.com/codeready-toolchain/host-operator/pull/758/files the CRD was generated correctly. But anyway, just wanted to mention it.
```suggestion
	// object
	// +optional
```

Reply:
I'll pull this commit down @alexeykazakov and re-run `make generate`

Reply:
`make generate` did not produce any changes 

```
gmontero ~/go/src/github.com/codeready-toolchain/api  (add-proxy-config)$ git log -3
commit c4878af3a4fc0ab73bea4fae648eadc04b8a3c51 (HEAD -> add-proxy-config, origin/add-proxy-config)
Author: Gabe Montero <gmontero@redhat.com>
Date:   Fri Mar 3 16:52:18 2023 -0500

    Update api/v1alpha1/proxyplugin_types.go
    
    Co-authored-by: Alexey Kazakov <alkazako@redhat.com>

commit ba9ecad6b1ee86c6b1a18ec544f65aae3c3d547a
Author: gabemontero <gmontero@redhat.com>
Date:   Thu Mar 2 13:51:00 2023 -0500

    make generate output

commit a46b3ff9e5992fa96a337237f3fe704483e53a49
Author: gabemontero <gmontero@redhat.com>
Date:   Thu Mar 2 13:50:40 2023 -0500

    add proxy plugin crd
gmontero ~/go/src/github.com/codeready-toolchain/api  (add-proxy-config)$ make generate 
rm -rf config/
Re-generating the deepcopy go file & the Toolchain CRD files... 
/home/gmontero/go/src/github.com/codeready-toolchain/api/bin/controller-gen crd \
object paths="./..." output:crd:artifacts:config=config/crd/bases
re-generating the openapi go file...
/home/gmontero/go/src/github.com/codeready-toolchain/api/bin/openapi-gen --input-dirs ./api/v1alpha1/ \
--output-package github.com/codeready-toolchain/api/api/v1alpha1 \
--output-file-base zz_generated.openapi \
--go-header-file=make/go-header.txt
API rule violation: names_match,github.com/codeready-toolchain/api/api/v1alpha1,RegistrationServiceVerificationConfig,AWSSenderID
Dispatching CRD files in the 'host-operator' and 'member-operator' repositories...
# Now let's remove the CRDs from config/crd/bases directory
ls: cannot access 'config/crd/bases/*yaml': No such file or directory
Dispatch successfuly finished \o/
cp ../registration-service/deploy/registration-service.yaml ../host-operator/deploy/registration-service/registration-service.yaml
gmontero ~/go/src/github.com/codeready-toolchain/api  (add-proxy-config)$ git status 
On branch add-proxy-config
nothing to commit, working tree clean
gmontero ~/go/src/github.com/codeready-toolchain/api  (add-proxy-config)$
```